nexuse2e-webapp/src/main/frontend/src/app/data/data.service.ts

```diff
@@ -47,6 +44,48 @@ export class DataService {
     return this.get<string>("/machine-name");
   }
 
+  buildFilterParams(activeFilters: ActiveFilter[]) {
+    let httpParams = new HttpParams();
+    for (const filter of activeFilters) {
+      if (filter.value) {
+        httpParams = httpParams.append(filter.fieldName, filter.value);
+      }
+    }
+    return httpParams;
+  }
+
+  getMessages(pageIndex: number, itemsPerPage: number, activeFilters: ActiveFilter[]): Promise<Message[]> {
+    let params = this.buildFilterParams(activeFilters);
+    params = params.append("pageIndex", String(pageIndex));
+    params = params.append("itemsPerPage", String(itemsPerPage));
+    return this.http
+      .get<Message[]>(this.API_URL + "/messages", { params: params })
+      .toPromise();
```

Comment:
Should we extend the `get` class method to take parameters?

Reply:
I'm not sure how to implement caching with the parameters, that's why I didn't implement this. Would you compare the path first, and then the parameters?

Reply:
Can we simply use the path concat with a JSON.stingfy of the params as cache key?